app/api/csv/arrangeThesauri.ts

```diff
@@ -59,23 +59,33 @@ type ThesauriValueData = {
   thesauriIdToExistingValues: Map<string, Set<string>>;
   thesauriIdToNewValues: Map<string, Set<string>>;
   thesauriIdToNormalizedNewValues: Map<string, Set<string>>;
+  thesauriIdToGroups: Map<string, Set<string>>;
 };
 
 const setupIdValueMaps = (allRelatedThesauri: WithId<ThesaurusSchema>[]): ThesauriValueData => {
   const thesauriIdToExistingValues = new Map();
   const thesauriIdToNewValues = new Map();
   const thesauriIdToNormalizedNewValues = new Map();
+  const thesauriIdToGroups = new Map();
 
   allRelatedThesauri.forEach(t => {
     const id = t._id.toString();
     const a = flatThesaurusValues(t, true);
     const thesaurusValues = a.map(v => normalizeThesaurusLabel(v.label));
+    const thesaurusGroups =
+      t.values?.filter(v => v.values)?.map(v => normalizeThesaurusLabel(v.label)) || [];
```

Comment:
I think the second optional chaining is not needed. `filter` will at least return an empty array.

Reply:
Thanks, corrected.